scaffolding-chef/lib/scaffolding.sh

```diff
@@ -52,8 +52,11 @@ export SSL_CERT_FILE="{{pkgPathFor "core/cacerts"}}/ssl/cert.pem"
 cd {{pkg.path}}
 
 exec 2>&1
+while true; do
 chef-client -z -l {{cfg.log_level}} -c $pkg_svc_config_path/client-config.rb --once
-exec chef-client -z -i {{cfg.interval}} -s {{cfg.splay}} -l {{cfg.log_level}} -c $pkg_svc_config_path/client-config.rb
+sleep {{cfg.interval}}
+sleep {{cfg.splay}}
```

Comment:
Splay is something random, isn't it?

Reply:
It totally is, and I need to figure a quick way out to use it. Doing $(()) math with {{cfg.values}} can be problematic.

Reply:
It _is_ random actually, no? https://github.com/chef/chef/blob/7c099ce3862ee4c11b8549d535a4a40798979a08/lib/chef/application/solo.rb#L336

Reply:
@jeremymv2 If we were using the ruby process, it would be random. Instead we need to generate our own random number in bash (I'm figuring out a way to do it with /dev/urandom) and feed it to sleep, in order to achieve our goal of being respectful to system resources.

Reply:
👍

Reply:
This is an interesting approach: https://bytefreaks.net/gnulinux/bash/bash-get-random-number-that-belongs-in-a-range

Reply:
$RANDOM is a bad approach because it is limited to 0 - 32767, and the randomness is rather poor.

Using /dev/urandom would be a pain the butt for a range function.

I decided to go with `shuf`, which is included in coreutils. If this approach turns out not to be good enough, then we can revisit.